problem_builder/tests/integration/test_slider.py

```diff
@@ -0,0 +1,147 @@
+# -*- coding: utf-8 -*-
+#
+# Copyright (c) 2014-2015 Harvard, edX & OpenCraft
+#
+# This software's license gives you freedom; you can copy, convey,
+# propagate, redistribute and/or modify this program under the terms of
+# the GNU Affero General Public License (AGPL) as published by the Free
+# Software Foundation (FSF), either version 3 of the License, or (at your
+# option) any later version of the AGPL published by the FSF.
+#
+# This program is distributed in the hope that it will be useful, but
+# WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU Affero
+# General Public License for more details.
+#
+# You should have received a copy of the GNU Affero General Public License
+# along with this program in a file in the toplevel directory called
+# "AGPLv3".  If not, see <http://www.gnu.org/licenses/>.
+#
+
+# Imports ###########################################################
+
+from .base_test import ProblemBuilderBaseTest, MentoringAssessmentBaseTest, CORRECT, GetChoices
+
+
+# Classes ###########################################################
+
+
+class SliderBlockTestMixins(object):
+    """ Mixins for testing slider blocks. Assumes only one slider block is on the page. """
+
+    def get_slider_value(self):
+        return int(self.browser.execute_script("return $('.pb-slider-range').val()"))
+
+    def set_slider_value(self, val):
+        self.browser.execute_script("$('.pb-slider-range').val(arguments[0]).change()", val)
+
+
+class SliderBlockTest(SliderBlockTestMixins, ProblemBuilderBaseTest):
+    """
+    Tests for the SliderBlock inside a normal Problem Builder block.
+    """
+    def test_simple_flow(self):
+        """ Test a regular Problem Builder block containing one slider """
+        pb_wrapper = self.load_scenario("slider_problem.xml", {"include_mcq": False})
+        self.wait_for_init()
+        # The initial value should be 50 and submit should be enabled since 50 is a valid value:
+        self.assertTrue(self.submit_button.is_enabled())
+        self.assertEqual(self.get_slider_value(), 50)
+        self.expect_checkmark_visible(False)
+        # Set the value to 75:
+        self.set_slider_value(75)
+        self.assertEqual(self.get_slider_value(), 75)
+        self.click_submit(pb_wrapper)
+        # Now, we expect submit to be disabled and the checkmark to be visible:
+        self.expect_checkmark_visible(True)
+        self.assertFalse(self.submit_button.is_enabled())
+        # Now change the value, and the button/checkmark should reset:
+        self.set_slider_value(45)
+        self.assertTrue(self.submit_button.is_enabled())
+        self.expect_checkmark_visible(False)
+        # Now reload the page:
+        self.browser.execute_script("$(document).html(' ');")
```

Comment:
@bradenmacdonald Why do we need this line? Just asking because I'd like to improve my understanding of integration tests :)